python/cuml/test/test_linear_model.py

```diff
@@ -38,17 +39,25 @@
 from sklearn.model_selection import train_test_split
 
 
-def make_regression_dataset(datatype, nrows, ncols, n_info):
+def _make_regression_dataset_uncached(nrows, ncols, n_info):
     X, y = make_regression(
         n_samples=nrows, n_features=ncols, n_informative=n_info, random_state=0
     )
-    X = X.astype(datatype)
-    y = y.astype(datatype)
-    X_train, X_test, y_train, y_test = train_test_split(
-        X, y, train_size=0.8, random_state=10
-    )
+    return train_test_split(X, y, train_size=0.8, random_state=10)
 
-    return X_train, X_test, y_train, y_test
+
+@lru_cache(4)
```

Comment:
great idea! I think it would be a good idea to generalize this for the whole test suite, what do you think? Could you create a github issue if you agree? (I think it should be its own PR)

Reply:
Before working on this PR, I would have said definitely, but I added caching in some other "obvious" places and it actually slowed us down. I'll throw up an issue to review possible places where caching would help us, though.

Reply:
Issue #3080 created.